src/gtk/textctrl.cpp

```diff
@@ -124,6 +124,41 @@ static void wxGtkTextApplyTagsFromAttr(GtkWidget *text,
         }
     }
 
+    if( attr.HasFontUnderlinedWithEffect() )
+    {
+        GtkTextTag *tag1;
+        PangoUnderline pg_style;
+        if( !(attr.GetUnderlinedType() == wxTEXT_ATTR_UNDERLINE_NONE || attr.GetUnderlinedType() == wxTEXT_ATTR_UNDERLINE_SOLID ) )
+        {
+            if( attr.GetUnderlinedType() == wxTEXT_ATTR_UNDERLINE_NONE )
+                pg_style = PANGO_UNDERLINE_NONE;
+            if( attr.GetUnderlinedType() == wxTEXT_ATTR_UNDERLINE_SOLID )
+                pg_style = PANGO_UNDERLINE_SINGLE;
+            if( attr.GetUnderlinedType() == wxTEXT_ATTR_UNDERLINE_DOUBLE )
+                pg_style = PANGO_UNDERLINE_DOUBLE;
+            if( attr.GetUnderlinedType() == wxTEXT_ATTR_UNDERLINE_WAVE )
+                pg_style = PANGO_UNDERLINE_ERROR;
+        }
+
+        g_snprintf(buf, sizeof(buf), "WXFONTUNDERLINEWITHEFFECT");
+        tag = gtk_text_tag_table_lookup( gtk_text_buffer_get_tag_table( text_buffer ),
+                                         buf );
+        if (!tag)
+        {
+            tag = gtk_text_buffer_create_tag( text_buffer, buf,
+                                              "underline-set", TRUE,
+                                              "underline", pg_style,
+                                              NULL );
+            if( gtk_check_version( 3, 16, 0 ) )
```

Comment:
```suggestion
            if ( gtk_check_version( 3, 16, 0 ) )
```
Please use the usual style.